src/libraries/System.Net.Ping/src/System/Net/NetworkInformation/IcmpV6MessageConstants.cs

```diff
@@ -77,11 +77,11 @@ public static IPStatus MapV6TypeToIPStatus(byte type, byte code)
                     switch ((IcmpV6TimeExceededCode)code)
                     {
                         case IcmpV6TimeExceededCode.HopLimitExceeded:
-                            return IPStatus.TimeExceeded;
+                            return IPStatus.TtlExpired;
```

Comment:
we can perhaps eliminate this whole branch and fall-back to default.

Reply:
I have removed the branch. 
I have also taken the liberty, since we are already making a change here anyway, to switch to Switch Expressions, since these are more readable in my opinion.